controllers/gitrepository_controller.go

```diff
@@ -756,14 +757,22 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 	gitCtx, cancel := context.WithTimeout(ctx, obj.Spec.Timeout.Duration)
 	defer cancel()
 
-	var gitReader git.RepositoryReader
+	var gitReader repository.Reader
 	var err error
 
 	switch gitImplementation {
 	case sourcev1.LibGit2Implementation:
-		gitReader, err = libgit2.NewClient(dir, authOpts)
+		clientOpts := []libgit2.ClientOption{libgit2.WithDiskStorage()}
+		if authOpts.Transport == git.HTTP {
+			clientOpts = append(clientOpts, libgit2.WithInsecureCredentialsOverHTTP())
+		}
+		gitReader, err = libgit2.NewClient(dir, authOpts, clientOpts...)
 	case sourcev1.GoGitImplementation:
-		gitReader, err = gogit.NewClient(dir, authOpts)
+		clientOpts := []gogit.ClientOption{gogit.WithDiskStorage()}
+		if authOpts.Transport == git.HTTP {
+			clientOpts = append(clientOpts, gogit.WithInsecureCredentialsOverHTTP())
```

Comment:
As a follow-up PR, the use of `WithInsecureCredentialsOverHTTP()` needs to be guarded by the `--insecure-allow-http` flag, as per [RFC0004](https://github.com/fluxcd/flux2/tree/main/rfcs/0004-insecure-http).